### PR TITLE
Email: template enhance for log analysis

### LINKED ISSUE #99: ceph-jenkins email template
* have suite info and log link
* possible fix in gitlab JJB ?

### LINKED ISSUE #100: Improve cephci email template
`http://post-office.corp.redhat.com/archives/cephci/2020-April/msg00196.html

@veera-raghava-reddy   
1. Subject - Can we include compose and suite name 
CEPHCI >> RHCEPH-4.1-RHEL-8-20200421.ci.0 >> sanity_containerized_ceph_ansible
2. Duration - Format in HH:MM:SS and total time at the end.
3. Test Name - Is it suite name, for few tests  same name "Versioning Tests"
  - If Test name is linked to Polarion test can we add a link to test
#95
4. In subject can we append the result - If [100% of tests Pass - Pass else - Fail, so it will help to prioritize on which test run mail to first look into.

### PR DESCRIPTION
**Pull request by XDsubhash**
* Resolves #100 
*  https://gitlab.cee.redhat.com/ceph/rhcs-jenkins-jobs/-/merge_requests/326 Resolves #99 
* cephfs test failures because of py2 dependency https://github.com/red-hat-storage/cephci/commit/ef51917d92fea88c34a2f99b15d0514c5a4392ce

### CODE REVIEW

**Comment by XDsubhash** [comment]
All test logs located here: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1594269121096

**Comment by XDsubhash** [comment]
@rakeshgm @yogesh-mane ,looks like py2 pip is not packaged in official software repositories of CentOS/RHEL still.
https://access.redhat.com/solutions/1519803
shud i revert the https://github.com/red-hat-storage/cephci/commit/ef51917d92fea88c34a2f99b15d0514c5a4392ce ?or is it better to fix in cephfs suite?as it seems to be the only suite which has some py2 dependency code

File: run.py [hunk 1]
```diff
@@ -577,18 +581,23 @@ def run(args):
...
         create_xunit_results(suite_name, tcs, run_dir)
-    url_base = "http://magna002.ceph.redhat.com/cephci-jenkins"
-    run_dir_name = run_dir.split('/')[-1]
     print("\nAll test logs located here: {base}/{dir}".format(base=url_base, dir=run_dir_name))
     print_results(tcs)
     send_to_cephci = post_results or post_to_report_portal
-    email_results(tcs, run_id, send_to_cephci)
+    run_end_time = datetime.datetime.now()
+    total_run_time = (datetime.datetime.now() - run_start_time)
+    total_time = """'''Start Time:''' {a} '''End Time:''' {b} '''Total duration:'''{c} \
```
Comment (sunilkumarn417) [inline run.py:598]: I would suggest individual row for Start/End/Total time/duration.

**Comment by rakeshgm** [comment]
lets remove it once the suite is moved to py3. I feel its a harmless code. @yogesh-mane please migrate the remaining FS suite to py3. @XDsubhash what do you think ?

File: run.py (hunk 1 above)
Comment (XDsubhash) [reply run.py:598]: i believe the general information table is already big enuf with 11 rows( gen info on par with actual test suite TC rows)

**Comment by XDsubhash** [comment]
@rakeshgm it was causing install pre-req task to fail sometimes when the repo link is down,hence removed it, frequent fedora releases is also a task to track when the repo link changes every few months.This would initiate py2 to py3 conversion and remove py2 packages installing as mentioned in # TODO

Better the function be included in cephfs util or one liner to pick form epel and install in cephfs test script where required.

**Comment by yogesh-mane** [comment]
@rakeshgm, me & @harishmunjulur are working on migration to py3 of fs suite.
Currently we are only seeing pip failures, @harishmunjulur is working on it.

**Comment by XDsubhash** [comment]
can we merge this ? as gitlab pr(merged) is dependent on this

Whole framework and most suites are moved to py3.It wouldnt be good to include py2 workaround in common install pre req task.
Focus groups can include in the component utils if needed.
@rakeshgm @sunilkumarn417  what do you think?

**Comment by rakeshgm** [comment]
@XDsubhash I agree, we can put the install py2/py3 changes in specific suites. we can merge it.

**Pull request merged**